public final class twitter4j.internal.async.DispatcherFactory extends java.lang.Object
{
    private twitter4j.conf.Configuration conf;
    private java.lang.String dispatcherImpl;

    public void <init>()
    {
        twitter4j.internal.async.DispatcherFactory $r0;
        twitter4j.conf.Configuration $r1;

        $r0 := @this: twitter4j.internal.async.DispatcherFactory;

        $r1 = staticinvoke <twitter4j.conf.ConfigurationContext: twitter4j.conf.Configuration getInstance()>();

        specialinvoke $r0.<twitter4j.internal.async.DispatcherFactory: void <init>(twitter4j.conf.Configuration)>($r1);

        return;
    }

    public void <init>(twitter4j.conf.Configuration)
    {
        twitter4j.internal.async.DispatcherFactory $r0;
        twitter4j.conf.Configuration $r1;
        java.lang.String $r2;

        $r0 := @this: twitter4j.internal.async.DispatcherFactory;

        $r1 := @parameter0: twitter4j.conf.Configuration;

        specialinvoke $r0.<java.lang.Object: void <init>()>();

        $r2 = interfaceinvoke $r1.<twitter4j.conf.Configuration: java.lang.String getDispatcherImpl()>();

        $r0.<twitter4j.internal.async.DispatcherFactory: java.lang.String dispatcherImpl> = $r2;

        $r0.<twitter4j.internal.async.DispatcherFactory: twitter4j.conf.Configuration conf> = $r1;

        return;
    }

    public twitter4j.internal.async.Dispatcher getInstance()
    {
        twitter4j.internal.async.DispatcherFactory $r0;
        java.lang.InstantiationException $r1;
        java.lang.String $r2;
        java.lang.Class $r3;
        java.lang.Class[] $r4;
        java.lang.reflect.Constructor $r5;
        java.lang.Object[] $r6;
        twitter4j.conf.Configuration $r7;
        java.lang.Object $r8;
        twitter4j.internal.async.Dispatcher $r9;
        java.lang.AssertionError $r10;
        java.lang.IllegalAccessException $r11;
        java.lang.ClassNotFoundException $r12;
        java.lang.ClassCastException $r13;
        java.lang.NoSuchMethodException $r14;
        java.lang.reflect.InvocationTargetException $r15;

        $r0 := @this: twitter4j.internal.async.DispatcherFactory;

        $r2 = $r0.<twitter4j.internal.async.DispatcherFactory: java.lang.String dispatcherImpl>;

     label01:
        $r3 = staticinvoke <java.lang.Class: java.lang.Class forName(java.lang.String)>($r2);

     label02:
        $r4 = newarray (java.lang.Class)[1];

     label03:
        $r4[0] = class "twitter4j/conf/Configuration";

        $r5 = virtualinvoke $r3.<java.lang.Class: java.lang.reflect.Constructor getConstructor(java.lang.Class[])>($r4);

     label04:
        $r6 = newarray (java.lang.Object)[1];

        $r7 = $r0.<twitter4j.internal.async.DispatcherFactory: twitter4j.conf.Configuration conf>;

     label05:
        $r6[0] = $r7;

        $r8 = virtualinvoke $r5.<java.lang.reflect.Constructor: java.lang.Object newInstance(java.lang.Object[])>($r6);

     label06:
        $r9 = (twitter4j.internal.async.Dispatcher) $r8;

     label07:
        return $r9;

     label08:
        $r1 := @caughtexception;

        $r10 = new java.lang.AssertionError;

        specialinvoke $r10.<java.lang.AssertionError: void <init>(java.lang.Object)>($r1);

        throw $r10;

     label09:
        $r11 := @caughtexception;

        $r10 = new java.lang.AssertionError;

        specialinvoke $r10.<java.lang.AssertionError: void <init>(java.lang.Object)>($r11);

        throw $r10;

     label10:
        $r12 := @caughtexception;

        $r10 = new java.lang.AssertionError;

        specialinvoke $r10.<java.lang.AssertionError: void <init>(java.lang.Object)>($r12);

        throw $r10;

     label11:
        $r13 := @caughtexception;

        $r10 = new java.lang.AssertionError;

        specialinvoke $r10.<java.lang.AssertionError: void <init>(java.lang.Object)>($r13);

        throw $r10;

     label12:
        $r14 := @caughtexception;

        $r10 = new java.lang.AssertionError;

        specialinvoke $r10.<java.lang.AssertionError: void <init>(java.lang.Object)>($r14);

        throw $r10;

     label13:
        $r15 := @caughtexception;

        $r10 = new java.lang.AssertionError;

        specialinvoke $r10.<java.lang.AssertionError: void <init>(java.lang.Object)>($r15);

        throw $r10;

        catch java.lang.InstantiationException from label01 to label02 with label08;
        catch java.lang.InstantiationException from label03 to label04 with label08;
        catch java.lang.InstantiationException from label05 to label06 with label08;
        catch java.lang.IllegalAccessException from label01 to label02 with label09;
        catch java.lang.IllegalAccessException from label03 to label04 with label09;
        catch java.lang.IllegalAccessException from label05 to label06 with label09;
        catch java.lang.ClassNotFoundException from label01 to label02 with label10;
        catch java.lang.ClassNotFoundException from label03 to label04 with label10;
        catch java.lang.ClassNotFoundException from label05 to label06 with label10;
        catch java.lang.ClassCastException from label01 to label02 with label11;
        catch java.lang.ClassCastException from label03 to label04 with label11;
        catch java.lang.ClassCastException from label05 to label07 with label11;
        catch java.lang.NoSuchMethodException from label01 to label02 with label12;
        catch java.lang.NoSuchMethodException from label03 to label04 with label12;
        catch java.lang.NoSuchMethodException from label05 to label06 with label12;
        catch java.lang.reflect.InvocationTargetException from label01 to label02 with label13;
        catch java.lang.reflect.InvocationTargetException from label03 to label04 with label13;
        catch java.lang.reflect.InvocationTargetException from label05 to label06 with label13;
    }
}
